lib/routes/master/widgets/punkapi_card.dart

```diff
@@ -1,28 +1,49 @@
 import 'package:crossplatformbeers/models/beer.dart';
 import 'package:flutter/material.dart';
 
-class PunkApiCard extends StatelessWidget {
-  static const gestureDetectorKey = Key('gestureDetectorKey');
-
+class PunkApiCard extends StatefulWidget {
   final Beer beer;
   final SelectedBeer onBeerSelected;
+  final AddBeerToFavorite onAddToFavorite;
+  final RemoveBeerToFavorite onRemoveToFavorite;
+  final bool isInFavorite;
 
-  const PunkApiCard({
-    Key key,
-    @required this.beer,
-    this.onBeerSelected,
-  })  : assert(beer != null),
+  const PunkApiCard(
+      {Key key,
+      @required this.beer,
+      this.onBeerSelected,
+      this.onAddToFavorite,
+      this.onRemoveToFavorite,
+      this.isInFavorite})
+      : assert(beer != null),
         super(key: key);
 
+  @override
+  _PunkApiCardState createState() => _PunkApiCardState();
+}
+
+class _PunkApiCardState extends State<PunkApiCard> {
+  static const gestureDetectorKey = Key('gestureDetectorKey');
+
+  bool _isInFavorite = false;
+
+  @override
+  void initState() {
+    super.initState();
+    setState(() {
```

Comment:
Tu as besoin de faire un setState pour le initState ?
"Importance of Calling SetState inside initState - Stack Overflow" https://stackoverflow.com/questions/53363774/importance-of-calling-setstate-inside-initstate#:~:text=The%20only%20time%20you%20can,updated%20to%20trigger%20a%20rebuild.